src/wolff_producer.erl

```diff
@@ -275,11 +299,39 @@ handle_cast(_Cast, St) ->
 code_change(_OldVsn, St, _Extra) ->
   {ok, St}.
 
-terminate(_, #{replayq := Q}) ->
-  ok = replayq:close(Q);
-terminate(_, _) ->
+terminate(_, State = #{replayq := Q}) ->
+  ok = replayq:close(Q),
+  clear_gauges(State, Q),
+  ok;
+terminate(_, State) ->
+  clear_gauges(State, undefined),
```

Comment:
maybe just call `wolff_metrics:queuing_set(Config, 0),` here because `undefined` is not a `replayq:q()`